lllc/main.cpp

```diff
@@ -52,10 +53,20 @@ void version()
 	exit(0);
 }
 
+void setEnv() {
+	std::setlocale(LC_ALL, "C");
+#if !defined(WIN32) && !defined(MAC_OSX) && !defined(__FreeBSD__) && !defined(__OpenBSD__)
+	if (!std::setlocale(LC_ALL, "")) {
+		setenv("LC_ALL", "C", 1);
```

Comment:
@chriseth Thank you for reviewing my pull request. These lines are important to ensure the program behaves in the least surprising way. Calling setlocale(LC_ALL, "C") is not all that is required to fix the problem where the user specified default locale is _invalid_. 

The call to setlocale with the empty string allows us to test whether the environment setting is valid (return non-NULL), and if not (return NULL) then we modify the process environment to a sensible default. In the latter case the call to setenv will prevent the runtime error exception.